scio-core/src/main/scala/com/spotify/scio/coders/instances/BeamTypeCoders.scala

```diff
@@ -49,6 +57,17 @@ trait BeamTypeCoders {
 
   implicit def matchResultMetadataCoder: Coder[MatchResult.Metadata] =
     Coder.beam(MetadataCoderV2.of())
+
+  implicit def genericJsonCoder[T <: GenericJson: ClassTag]: Coder[T] =
+    Coder.xmap(Coder[String])(
+      str => DefaultDsonObjectParser.parseAndClose(new StringReader(str), ScioUtil.classOf[T]),
+      (s: T) => {
+        s.setFactory(new JacksonFactory)
+        s.toString
+      }
+    )
 }
 
-private[coders] object BeamTypeCoders extends BeamTypeCoders
+private[coders] object BeamTypeCoders extends BeamTypeCoders {
+  private lazy val DefaultDsonObjectParser = new JsonObjectParser(new JacksonFactory)
```

Comment:
Is `Dson` a typo or intentional? :O

Reply:
Yeah It's a new format i came up with :trollface: nice catch!